### DISCUSSION TITLE
Add ability to set fuseOptions in the JavaScript Select component

### DISCUSSION THREAD

**Discussion by ldraw-org** [filamentphp/filament#14743] [Ideas]
I posted this over on Discord but it's probably more useful here.

Choices.js uses Fuse.js for the search feature. Per the [Fuse.js documentation](https://www.fusejs.io/api/options.html#ignorelocation), the default search options only search the first 60 characters of the option label (or value if that option is set). This means that if you have an option label and the string you are searching for occurs after 60 character mark, the search feature will fail to find any result for that option. This default can be modified by setting fuseOptions on the Choices control.

Here's a pen highlighting the above:
[https://codepen.io/billthefish/pen/BaXqqQN](https://codepen.io/billthefish/pen/BaXqqQN)

While I can certainly create my own Select class/plugin with a property, blade template, and JS that allows these options to be set, I think it would be useful if this functionality was built into the core Select components.

I also did some poking around and found this issue:
#10550 
I think the error this user was experiencing is a result of the above limitation and not a bug in Choice,js

**Comment by areichardt-gedisa** [comment]
I would found this very useful too.

The fuzzy search of Fuse.js is not very useful in our case and the only way to avoid it, is using a custom search via ajax calls.
It would be great if we could adjust the threshold of Fuse.js directly.

Would it make sense to provide a PR for be able to adjust the Fuse.js options?
